var/spack/repos/builtin/packages/dray/package.py

```diff
@@ -64,34 +71,34 @@ class Dray(Package, CudaPackage):
     depends_on("conduit~shared", when="~shared")
     depends_on("conduit+shared", when="+shared")
 
-    depends_on("apcomp~shared+openmp+mpi", when="~shared+openmp+mpi")
-    depends_on("apcomp+shared+openmp+mpi", when="+shared+openmp+mpi")
-    depends_on("apcomp~shared~openmp+mpi", when="~shared~openmp+mpi")
-    depends_on("apcomp+shared~openmp+mpi", when="+shared~openmp+mpi")
-    depends_on("apcomp~shared+openmp~mpi", when="~shared+openmp~mpi")
-    depends_on("apcomp+shared+openmp~mpi", when="+shared+openmp~mpi")
-    depends_on("apcomp~shared~openmp~mpi", when="~shared~openmp~mpi")
-    depends_on("apcomp+shared~openmp~mpi", when="+shared~openmp~mpi")
+    depends_on("apcomp~mpi", when="~mpi")
+    depends_on("apcomp+mpi", when="+mpi")
+    depends_on("apcomp~openmp", when="~openmp")
+    depends_on("apcomp+openmp", when="+openmp")
+    depends_on("apcomp~shared", when="~shared")
+    depends_on("apcomp+shared", when="+shared")
 
     depends_on("raja@:0.13", when="@:0.1.6")
-    depends_on("raja+cuda~openmp+shared", when="+cuda~openmp+shared")
-    depends_on("raja+cuda+openmp+shared", when="+cuda+openmp+shared")
-    depends_on("raja+cuda~openmp~shared", when="+cuda~openmp~shared")
-    depends_on("raja+cuda+openmp~shared", when="+cuda+openmp~shared")
-
-    depends_on("raja~cuda~openmp+shared", when="~cuda~openmp+shared")
-    depends_on("raja~cuda+openmp+shared", when="~cuda+openmp+shared")
-    depends_on("raja~cuda~openmp~shared", when="~cuda~openmp~shared")
-    depends_on("raja~cuda+openmp~shared", when="~cuda+openmp~shared")
+    depends_on("raja~cuda", when="~cuda")
+    depends_on("raja+cuda", when="+cuda")
+    propagate_cuda_arch('raja')
+    depends_on("raja~shared", when="~shared")
+    depends_on("raja+shared", when="+shared")
+    depends_on("raja~openmp", when="~openmp")
+    depends_on("raja+openmp", when="+openmp")
 
     depends_on("umpire@:4.9", when="@:0.1.6")
-    depends_on("umpire+cuda+shared", when="+cuda+shared")
-    depends_on("umpire+cuda~shared", when="+cuda~shared")
+    # Only use umpire cuda if not shared.
+    depends_on("umpire+cuda", when="+cuda")
+    depends_on("umpire~cuda", when="~cuda")
+    depends_on("umpire+cuda~shared", when="+cuda+shared")
```

Comment:
This is a workaround for umpire since it does not build +cuda+shared, but it does build static libraries with PIC enabled. Ideally upstream umpire should be updated to allow building `+cuda+shared`.